File: app/src/main/java/com/qimian233/ztool/hook/modules/systemFramework/KeepRotation.java
package com.qimian233.ztool.hook.modules.systemFramework;

import com.qimian233.ztool.hook.base.BaseHookModule;

import io.github.libxposed.api.XposedModuleInterface;

import java.lang.reflect.Method;

public class KeepRotation extends BaseHookModule {
    public static final String FEATURE_NAME = "keep_rotation";
    public static final String TARGET_PACKAGE = "system";

    public KeepRotation() {}

    public String getModuleName() { return FEATURE_NAME; }
    public String[] getTargetPackages() { return new String[]{TARGET_PACKAGE};}

    @Override
    public void handleLoadPackage(XposedModuleInterface.PackageLoadedParam param) throws Throwable {

    }

    @Override
    public void handleSystemServerStarting(XposedModuleInterface.SystemServerStartingParam param) throws Throwable {
        ClassLoader classLoader = param.getClassLoader();
        log("Hooking DisplayRotation.isRotationCts");
        try{
            Method method = classLoader.loadClass("com.zui.server.wm.ZuiDisplayRotation")
                    .getDeclaredMethod("isRotationCts");
            hookWithId(method, "is_rotation_cts", chain -> Boolean.TRUE);
            log("Hooked DisplayRotation.isRotationCts [OK]");
        } catch (Exception e) {
            logError("Error hooking DisplayRotation.isRotationCts", e);
        }
    }
}

File: app/src/main/java/com/qimian233/ztool/hook/modules/systemframework/KeepRotation.kt
package com.qimian233.ztool.hook.modules.systemframework

import com.qimian233.ztool.data.keys.PreferenceKeys
import com.qimian233.ztool.data.keys.ScopeKeys
import com.qimian233.ztool.hook.base.SystemHookModule
import io.github.libxposed.api.XposedModuleInterface.SystemServerStartingParam
import java.lang.Boolean
import kotlin.Array
import kotlin.String
import kotlin.Throwable
import kotlin.arrayOf

class KeepRotation : SystemHookModule() {
    override fun getModuleName(): String = PreferenceKeys.KEEP_ROTATION.name

    override fun getTargetPackages(): Array<String> = arrayOf(ScopeKeys.SYSTEM_SERVER.packageName)

    override fun handleSystemServerStarting(param: SystemServerStartingParam) {
        val classLoader = param.classLoader
        logger.info("Hooking DisplayRotation.isRotationCts")
        try {
            val method = classLoader.loadClass("com.zui.server.wm.ZuiDisplayRotation")
                .getDeclaredMethod("isRotationCts")
            hookWithId(method, "is_rotation_cts") { Boolean.TRUE }
            logger.info("Hooked DisplayRotation.isRotationCts [OK]")
        } catch (th: Throwable) {
            logger.error("Error hooking DisplayRotation.isRotationCts", th)
        }
    }
}
